File: Exploratory/AtomicityDec3/src/AtomicityDec3_c_FMI/environImpl.c
#include "Common.h"
// Protected: environImpl

// Variables and constants
int b_environImpl = 99;


// Subroutines
void environImpl_env_fmiDoStep()
{
// [Internal] This will block until the guard has been met (no nice way of doing this in OpenMP 2.5)
BOOL completed = FALSE; 	 // [Internal] Private to the thread by default
// [Internal] Try and get lock
// Translated code
b_environImpl = (b_environImpl + 1);
// [Internal] Set completed flag
completed = TRUE;
}

void environImpl_fmiGetInteger(int *p)
{
// [Internal] This will block until the guard has been met (no nice way of doing this in OpenMP 2.5)
BOOL completed = FALSE; 	 // [Internal] Private to the thread by default
// [Internal] Try and get lock
// Translated code
(*p)= b_environImpl ;
// [Internal] Set completed flag
completed = TRUE;
}

File: Exploratory/AtomicityDec3/src/atomicityDec3_java/environImpl.java
package atomicityDec3_java;
// Protected: environImpl

public class environImpl
{
	
	// Instance variables and constants
	protected static int b = 99;
	protected static boolean tmp = true;
	private static Object lock = new Object();
	
	public environImpl()
	{
		
	}
	
	// Subroutines
	public void fmiGetInteger(Pointer<Integer> b)
	{
		// [Internal] This will block until the guard has been met
		boolean completed = false; 	 // private to the thread by default
		// [Internal] Try and get lock
		synchronized(lock)
		{
			// Translated code
			// [Internal] Set completed flag
			completed = true;
		}
	}
	
	public void env_fmiDoStep()
	{
		// [Internal] This will block until the guard has been met
		boolean completed = false; 	 // private to the thread by default
		// [Internal] Try and get lock
		synchronized(lock)
		{
			// Translated code
			b = (b + 1);
			// [Internal] Set completed flag
			completed = true;
		}
	}
	
	public void fmiDoStep()
	{
		// [Internal] This will block until the guard has been met
		boolean completed = false; 	 // private to the thread by default
		// [Internal] Try and get lock
		synchronized(lock)
		{
			// Translated code
			b = (b + 1);
			System.out.println("b: " + b);
			// [Internal] Set completed flag
			completed = true;
		}
	}
	
}
